### PR TITLE
[Intel MKL] Fixing bfloat16 integration in MatMul and BatchMatMul for…

### PR DESCRIPTION
**Pull request by nhasabni**
… DNNL1.2

This PR corrects bfloat16 integration in MatMul and BatchMatMul kernels for Intel MKL backend.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc [hunk 1]
```diff
@@ -174,9 +174,29 @@ class BatchMatMulMkl : public OpKernel {
...
+      a = &lhs_reshaped(0, 0, 0);
+      b = &rhs_reshaped(0, 0, 0);
+      Scalar* c = &out_reshaped(0, 0, 0);
+      // TODO(nhasabni): Use appropriate cast instead of passing addresses of
+      // a,b and c.
+      MklCblasGemmBatch(CblasRowMajor, adj_x_, adj_y_, m_array, n_array,
+                        k_array, &a, lda_array, &b, ldb_array, &c, ldc_array, 1,
+                        group_size);
+    } else {
+      // MKL CBLAS API requires a, b, and c as array of pointers, where each
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_batch_matmul_op.cc:194]: `a_array`, `b_array`, and `c_array` are only used here. Please put lines 151-174 in `else` too.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc [hunk 2]
```diff
@@ -302,7 +323,7 @@ TF_CALL_double(REGISTER_BATCH_MATMUL_MKL_V2);
 TF_CALL_complex64(REGISTER_BATCH_MATMUL_MKL_V2);
 TF_CALL_complex128(REGISTER_BATCH_MATMUL_MKL_V2);
 
-#if defined(ENABLE_INTEL_MKLDNN_V1) && defined(ENABLE_INTEL_MKL_BFLOAT16)
+#if defined(ENABLE_MKLDNN_V1) && defined(ENABLE_INTEL_MKL_BFLOAT16)
 TF_CALL_bfloat16(REGISTER_BATCH_MATMUL_MKL);
 TF_CALL_bfloat16(REGISTER_BATCH_MATMUL_MKL_V2);
 #endif  // ENABLE_INTEL_MKLDNN_V1 && ENABLE_INTEL_MKL_BFLOAT16
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_batch_matmul_op.cc:329]: Nice catch! Please update here as well.
```suggestion
#endif  // ENABLE_MKLDNN_V1 && ENABLE_INTEL_MKL_BFLOAT16
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm sorry for the delay!

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_batch_matmul_op.cc:195]: Done. Thanks.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc (hunk 2 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_batch_matmul_op.cc]: Done.

**Review by penpornk — APPROVED** [review]
Thank you very much again!

**Comment by nhasabni** [comment]
Thanks for review!

**Pull request merged**